src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+            
+            selectedProblemList = new ArrayList<Problem>();
+            valuesList.forEach((prob) -> {selectedProblemList.add((Problem)prob);});
+        }
+	}
+
+	protected void updateNosLabel(List<Object> valuesList) {
+
+        if (valuesList == null || valuesList.size() == 0) {
+            selectedNosLabel.setText("None selected");
+            selectedNosLabel.setToolTipText("None selected");
+        } else {
+
+            selectedNosLabel.setText(valuesList.size() + " selected");
+            selectedNosLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            submissionSolutionList = new SubmissionSolutionList();
+
+            // update/load selected items into submissionSolutionList
+            valuesList.forEach((subSL) -> {
+                submissionSolutionList.add((SubmissionSampleLocation) subSL);
+            });
+
+        }
+    }
+
+	protected void clearTextArea() {
+		textArea.selectAll();
+		textArea.replaceSelection("");
+		linenumber = 0;
+	}
+
+	void addLineToTextArea(String s) {
+
+		linenumber++;
+		String fmtLineNumber = String.format("%4d", linenumber);
+
+		textArea.insert(fmtLineNumber + " " + hhMMSSformatter.format(new Date()) + " " + s + "\n", 0);
+//		textArea.append(s);
+	}
+
+	protected void selectNewCDP() {
+
+		File file = selectEntry("Select CDP");
+
+		File cdpDir = FileUtilities.findCDPConfigDirectory(file);
+
+		if (cdpDir != null) {
+		    updateCDPDirAndFields(cdpDir.getAbsolutePath());
+		} else {
+
+			int result = FrameUtilities.yesNoCancelDialog(this,
+					file.getAbsoluteFile() + " may not be a CDP directory, continue anyways?", "Select CDP");
+			if (result == JOptionPane.YES_OPTION) {
+	            updateCDPDirAndFields(file.getAbsolutePath());
+			}
+		}
+	}
+
+	/**
+	 * Select yaml file/entry.
+	 * 
+	 * @param dialogTitle
+	 * @return
+	 */
+	protected File selectEntry(String dialogTitle) {
+
+		JFileChooser chooser = new JFileChooser(cdpTextField.getText());
+//        chooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+		FileFilter filterYAML = new FileNameExtensionFilter("YAML document (*.yaml)", "yaml");
+		chooser.addChoosableFileFilter(filterYAML);
+
+		chooser.setAcceptAllFileFilterUsed(false);
+		chooser.setFileFilter(filterYAML);
+
+		int action = chooser.showOpenDialog(this);
+		chooser.setDialogTitle(dialogTitle);
+
+		switch (action) {
+		case JFileChooser.APPROVE_OPTION:
+			File file = chooser.getSelectedFile();
+			return file;
+		case JFileChooser.CANCEL_OPTION:
+		case JFileChooser.ERROR_OPTION:
+		default:
+			break;
+		}
+		return null;
+
+	}
+
+	void updateClientCDPPath(String path) {
+		ClientSettings settings = getContest().getClientSettings();
+		if (settings == null) {
+			settings = new ClientSettings(getContest().getClientId());
+		}
+		settings.put(CUSTOM_SUBMIT_SAMPLE_CDP_PATH, path);
+
+		getController().updateClientSettings(settings);
+	}
+
+	private void xlog(String string) {
+		System.out.println(string);
```

Comment:
This prints a message to stdout every time `xlog` is called -- which happens once when `setContestAndController()` is called, and then _once every time `showMessage()` is called_ and also once _for every file to be submitted each time `submitSelectedJudgesSolutions()` is called_.  I thought we agreed that such output to stdout should be highly avoided.  Since the output is _also_ going to the "output text area", it seems like this writing to stdout should be avoided.